modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1458,12 +1458,46 @@ public void scrollTo(int index) {
         if (cell != null) {
             scrollTo(cell);
         } else {
+            // see JDK-8197536
+            if (tryScrollOneCell(index, true)) {
+                return;
+            } else if (tryScrollOneCell(index, false)) {
+                return;
+            }
+
             adjustPositionToIndex(index);
             addAllToPile();
             requestLayout();
         }
     }
 
+    // will return true if scroll is successful
+    private boolean tryScrollOneCell(int targetIndex, boolean downOrRight) {
+        // if going down, cell diff is -1, because it will get the target cell index and check if previous
+        // cell is visible to base the position
+        int indexDiff = downOrRight ? -1 : 1;
```

Comment:
Remove space before '?'

Reply:
No, the space is consistent with our coding style.